### PR TITLE
owner init + matching based on cur house

### PR DESCRIPTION
**Pull request by ttillett13**
1. moved default matching to "1" so it's always first no matter how many methods we add
2. added test2.txt
3. added an owners initialization to init agent i to house i
4. added a matching which breaks ties based on how far down the list the agent's current house is (gives priority to agents who don't like their houses)

### CODE REVIEW

**Comment by bethrichardson** [comment]
Should we add tests for both the old initialization and matching strategies and these new ones so we can show that we support more than one of each?

**Comment by ttillett13** [comment]
> Should we add tests for both the old initialization and matching strategies and these new ones so we can show that we support more than one of each?

this is a good idea

File: src/main/java/edu/texas/social/computing/housing/initialization/OwnersInitialization.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        Map<House, Agent> owners = new HashMap<>();
+        for (int i=0; i<n; i++) {
+            House h = new House(i);
+            Agent a = agents.get(i);
+            owners.put(h, a);
+        }
+        return owners;
+    }
+
+    /*@Override
```
Comment (JavierPalomares90) [inline src/main/java/edu/texas/social/computing/housing/initialization/OwnersInitialization.java:28]: can we get rid of this commented out code?

File: src/main/java/edu/texas/social/computing/housing/matching/FairOwnerMatching.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+                // figure out index in prefs of agent's current house
+                int pref = NUM_AGENTS - 1;
+                for (int i=0; i<NUM_AGENTS; i++) {
+                    House p = a.preferences.get(i);
+                    if (a.equals(owners.get(p))) { // agent owns this house
+                        pref = i;
+                        break;
+                    }
+                }
+                if (pref > worstPref) {
```
Comment (JavierPalomares90) [inline src/main/java/edu/texas/social/computing/housing/matching/FairOwnerMatching.java:75]: what does this condition do? if agent a prefers this home more than any previous agent, assign the house to the agent? correct?

Comment (ttillett13) [reply src/main/java/edu/texas/social/computing/housing/matching/FairOwnerMatching.java:75]: it's keeping track of which agent prefers their current house the least. so worstPref is the worst preference that an agent has for their current house. so if my house is my 3rd preference and your house is your 4th preference then worstPref would be 4

**Pull request merged**